blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md

```diff
@@ -0,0 +1,212 @@
+
+1. Go to [the celo explorer page](https://explorer.celo.org)
+2. Type the name or the address of the contract in the search bar and click on the search icon.
+3. Once the contract is displayed, click on the "ABI" tab to see the contract's ABI
+4. You can then copy the ABI and use it in your code
+
+Also, you can use the following step to get your bytecode.
+
+1. Go to [the celo explorer page](https://explorer.celo.org)
+2. Search for your contract on celo explorer and copy the bytecode from the "Bytecode" section of the contract details.
+
+
+Let's estimate the gas cost for deploying the contract with following code
+
+``` py
+gas_estimate = w3.eth.estimate_gas({'from': address, 'data': bytecode})
+```
+
+We use the `estimate_gas` function of the `web3.py` eth module to estimate the amount of gas required to deploy the contract. The function takes a dictionary with the from address and the contract data (bytecode) as arguments. The gas estimate is then stored in the gas_estimate variable for use in the contract deployment transaction.
+Replace the address and bytecode with your contract address and bytecode respectively
+
+Now we can make the transaction, first step is building the transaction. Here's the code for building the transaction to deploy the contract:
+
+```py
+tx = {
+ 'from': address,
+ 'gas': gas_estimate,
+ 'gasPrice': Web3.to_wei('10', 'gwei'), 
+'nonce': w3.eth.get_transaction_count(address),
+ 'data': bytecode 
+}
+```
+
+then sign the transaction with the following code line
+
+```py
+signed_tx = w3.eth.account.sign_transaction(tx, private_key)
+```
+
+And finally we can send the transaction
+
+```py
+tx_hash = w3.eth.send_raw_transaction(signed_tx.rawTransaction)
+```
+
+Next step is to wait for the transaction to be mined, we’ll do that with the following code. Note that the `tx_hash` is the hash of the transaction we just sent.
+
+```py
+tx_receipt = w3.eth.wait_for_transaction_receipt(tx_hash)
+```
+
+The code will return a TransactionReceipt object that contains information about the transaction, such as the block number and gas used.
+
+Now that the transaction has been mined we can get the deployed contract address with following code
+
+```py
+contract_address = tx_receipt.contract_address
+```
+
+We can then generate a contract instance using `web3.py` and `ConciseContract`. The `ConciseContract` is not a part of the official Web3.py documentation, but rather a third-party library that builds on top of Web3.py to provide a simpler, more user-friendly interface for interacting with smart contracts. To use ConciseContract, you would need to install the web3-contract package, which provides the `ConciseContract`. You can install the it with the following command:
+```py
+pip install web3-contract
+```
+Now, we can generate the contract instance
+
+```py
+#First import the ConciseContract
+from web3.contract import ConciseContract
+MyContract = ConciseContract(w3.eth.contract(address=contract_address, abi=abi))
+```
```

Comment:
Move the installation process at the beginning and also the import process at the beginning